frontend/pages/Homepage/cards/Software/Software.tsx

```diff
@@ -221,29 +235,26 @@ const Software = ({
     }
   };
 
-  const onModalSoftwareQueryChange = debounce(
+  const onModalSoftwareQueryChange = useDebouncedCallback(
     async ({ pageIndex, searchQuery }: ITableQueryProps) => {
       setModalSoftwareSearchText(searchQuery);
 
       if (pageIndex !== modalSoftwarePageIndex) {
         setModalSoftwarePageIndex(pageIndex);
+      } else {
+        try {
+          // TODO: count loading UX?
+          const count = await softwareAPI.count(searchQuery);
+          setModalSoftwareCount(count);
+        } catch (err) {
+          // TODO: count errors UX?
+          console.log(err);
```

Comment:
I think for this we can update the text to say "Count failed". A good question for Noah related to the one above.